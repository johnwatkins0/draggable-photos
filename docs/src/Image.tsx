import * as React from 'react';

import DraggableSplitscreen from '../../src/draggable-splitscreen';

const LeftSide = () => (
  <img
    src="/react-draggable-splitscreen/docs/assets/left-side.png"
    alt="Left Side"
    style={{ width: '100%', height: 'auto' }}
  />
);

const RightSide = () => (
  <img
    src="/react-draggable-splitscreen/docs/assets/right-side.png"
    alt="Right Side"
    style={{ width: '100%', height: 'auto' }}
  />
);

const Image = () => <DraggableSplitscreen leftSide={<LeftSide />} rightSide={<RightSide />} />;

export default Image;
