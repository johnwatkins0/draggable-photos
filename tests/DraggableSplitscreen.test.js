import React from 'react';
import renderer from 'react-test-renderer';

import Image from '../example/src/Image';
import Text from '../example/src/Text';
import Video from '../example/src/Video';

test('Image-based element renders correctly.', () => {
  const tree = renderer.create(<Image />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Text-based element renders correctly.', () => {
  const tree = renderer.create(<Text />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Video-based element renders correctly.', () => {
  const tree = renderer.create(<Video />).toJSON();

  expect(tree).toMatchSnapshot();
});
