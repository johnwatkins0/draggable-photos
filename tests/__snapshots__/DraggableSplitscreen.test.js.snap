// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image-based element renders correctly. 1`] = `
<div
  className="DraggableSplitscreen sc-bdVaJa cQtrjW"
>
  <div
    className="DraggableSplitscreen__left-side sc-bwzfXH uTxCW"
  >
    <img
      alt="Left Side"
      src="/react-draggable-splitscreen/example/assets/left-side.png"
      style={
        Object {
          "height": "auto",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="DraggableSplitscreen__right-side sc-htpNat hkAeFy"
  >
    <img
      alt="Right Side"
      src="/react-draggable-splitscreen/example/assets/right-side.png"
      style={
        Object {
          "height": "auto",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="DraggableSplitscreen__handle react-draggable sc-bxivhb kRBqZl"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "touchAction": "none",
        "transform": "translate(0px,0px)",
      }
    }
    transform={null}
  />
</div>
`;

exports[`Text-based element renders correctly. 1`] = `
<div
  className="DraggableSplitscreen sc-bdVaJa cQtrjW"
>
  <div
    className="DraggableSplitscreen__left-side sc-bwzfXH uTxCW"
  >
    <div
      style={
        Object {
          "backgroundColor": "blue",
          "padding": "6rem",
        }
      }
    >
      <h2>
        <a
          href="#"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Left side
        </a>
      </h2>
    </div>
  </div>
  <div
    className="DraggableSplitscreen__right-side sc-htpNat hkAeFy"
  >
    <div
      style={
        Object {
          "backgroundColor": "red",
          "padding": "6rem",
          "textAlign": "right",
        }
      }
    >
      <h2>
        <a
          href="#"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Right side
        </a>
      </h2>
    </div>
  </div>
  <div
    className="DraggableSplitscreen__handle react-draggable sc-bxivhb kRBqZl"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "touchAction": "none",
        "transform": "translate(0px,0px)",
      }
    }
    transform={null}
  />
</div>
`;

exports[`Video-based element renders correctly. 1`] = `
<div
  className="DraggableSplitscreen sc-bdVaJa cQtrjW"
>
  <div
    className="DraggableSplitscreen__left-side sc-bwzfXH uTxCW"
  >
    <video
      autoPlay={true}
      loop={true}
      muted={true}
      src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
      style={
        Object {
          "height": "auto",
          "objectFit": "inherit",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="DraggableSplitscreen__right-side sc-htpNat hkAeFy"
  >
    <video
      autoPlay={true}
      loop={true}
      muted={true}
      src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4"
      style={
        Object {
          "height": "auto",
          "objectFit": "inherit",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="DraggableSplitscreen__handle react-draggable sc-bxivhb kRBqZl"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "touchAction": "none",
        "transform": "translate(0px,0px)",
      }
    }
    transform={null}
  />
</div>
`;
